spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1302,4 +1304,50 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("test cast utf8 to boolean as compatible with Spark") {
+    withSQLConf(
+      CometConf.COMET_ENABLED.key -> "true",
+      CometConf.COMET_EXEC_ALL_OPERATOR_ENABLED.key -> "true") {
+      withTable("test_table1", "test_table2", "test_table3", "test_table4") {
+        // Supported boolean values as true by both Arrow and Spark
+        val inputDF = Seq("t", "true", "y", "yes", "1", "T", "TrUe", "Y", "YES").toDF("c1")
+        inputDF.write.format("parquet").saveAsTable("test_table1")
+        val resultDF = this.spark
+          .table("test_table1")
+          .withColumn("converted", col("c1").cast(DataTypes.BooleanType))
+        val resultArr = resultDF.collectAsList().toList
+        resultArr.foreach(x => assert(x.get(1) == true))
```

Comment:
Besides this, we can use `checkSparkAnswerAndOperator` to check if the result is same as Spark.